benchmark/metadata/metadata_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"google.golang.org/grpc/metadata"
+	"testing"
+)
+
+func Benchmark_AddingMetadata_WithContextCopying(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		ctx := context.Background()
+		md, _ := metadata.FromOutgoingContext(ctx)
+		send := map[string]string{"k1": "v1", "k2": "v2"}
```

Comment:
This would be more efficiently written as: `newMD := metadata.Pairs()` to avoid allocating an extra map.

Reply:
I actually just included this benchmark as a comparison to what was happening in the original issue (#1390), in which case the original `md` would be populated`. Since that'll no longer be the preferred approach to appending metadata, I reckon this benchmark can be deleted during the squash - thoughts?

Reply:
I could go either way.  This is the "old way" of doing it, so it's a good idea to compare it to the "new way", but I don't think we need to keep it once the PR is submitted, either.

Reply:
👍  I'll keep it